### PR TITLE
sdk-wallet: fix names, move behavior (functions)

### PR DESCRIPTION
**Pull request by andreibancioiu**

### CODE REVIEW

File: sdk-wallet/user_wallet_provider.md [hunk 1]
```diff
@@ -28,6 +31,14 @@ class UserWalletProvider:
 
     // Should not throw.
     generate_mnemonic(): Mnemonic
+
+    // Should not throw.
+    validate_mnemonic(mnemonic: Mnemonic): bool
+
+    derive_secret_key_from_mnemonic(mnemonic: Mnemonic, address_index: int, passphrase: string): ISecretKey
```
Comment (andreibancioiu) [inline sdk-wallet/user_wallet_provider.md:38]: Moved here, from `mnemonic`. E.g. `bip39` primitives are now hidden inside the wallet provider, not inside `mnemonic`.

File: sdk-wallet/validator_wallet_provider.md [hunk 2]
```diff
@@ -21,4 +24,8 @@ class ValidatorWalletProvider:
     // Can throw:
     // - ErrInvalidPublicKey
```
Comment (bogdan-rosianu) [inline sdk-wallet/validator_wallet_provider.md:25]: `ErrInvalidPublicKey` -> `ErrInvalidPublicKeyBytes`, maybe? Since the bytes are invalid

**Review by bogdan-rosianu — APPROVED** [review]

File: sdk-wallet/validator_wallet_provider.md (hunk 2 above)
Comment (andreibancioiu) [reply sdk-wallet/validator_wallet_provider.md]: Fixed.

**Comment by andreibancioiu** [comment]
> I think it might be worth to add an `IWallet` specification, since besides the `generate_mnemonic` method they share the same methods and signatures. Either in this PR or in another one, whatever you may fancy.

Indeed, added `IWalletProvider` and `IMnemonicComputer`.

File: sdk-wallet/user_wallet_provider.md (hunk 1 above)
Comment (andreibancioiu) [reply sdk-wallet/user_wallet_provider.md]: Later - extracted to `MnemonicComputer` (a bit similar in design to `AddressComputer`, `TransactionComputer`).

File: sdk-wallet/interfaces.md [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    create_secret_key_from_bytes(data: bytes): ISecretKey
+    create_public_key_from_bytes(data: bytes): IPublicKey
+    compute_public_key_from_secret_key(secret_key: ISecretKey): IPublicKey
+```
+
+```
+interface IMnemonicComputer:
+    generate_mnemonic(): Mnemonic
+    validate_mnemonic(mnemonic: Mnemonic): bool
+    derive_secret_key_from_mnemonic(mnemonic: Mnemonic, address_index: int, passphrase: string): ISecretKey
```
[the remark this answers is not in the document]
Comment (andreibancioiu) [reply sdk-wallet/interfaces.md:21]: Double-checked with @ccorcoveanu, we should rename `passphrase` to `password`.

Secret key derivation from mnemonic supports an additional, optional password:
https://github.com/multiversx/mx-sdk-js-wallet/blob/main/src/mnemonic.ts#L36

For reference:
https://help.blockstream.com/hc/en-us/articles/8712301763737-What-is-a-BIP39-passphrase-

[the remark this answers is not in the document]
Comment (ccorcoveanu) [reply sdk-wallet/interfaces.md:21]: Actually, @andreibancioiu I double-checked BIP39 specs, both password and passphrase are used in the description, but passphrase is predominant. I would say we should actually let it as passphrase

**Review by iulianpascalau — APPROVED** [review]

File: sdk-wallet/interfaces.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (andreibancioiu) [reply sdk-wallet/interfaces.md:21]: Added extra description.

**Review by iulianpascalau — APPROVED** [review]

**Review by bogdan-rosianu — APPROVED** [review]

**Review by ccorcoveanu — APPROVED** [review]

File: sdk-wallet/interfaces.md [hunk 4]
```diff
@@ -0,0 +1,22 @@
+## Interfaces
+
+For languages that support **structural typing** (e.g. Go, Python, TypeScript), the following interfaces should not be _exported_, since we'd like to encourage client applications to define their own interfaces, as needed.
+
+For languages that only support **nominal typing** (e.g. C#), these interfaces can be _exported_.
+
+```
+interface IWalletProvider:
```
[the remark this answers is not in the document]
Comment (AdoAdoAdo) [reply sdk-wallet/interfaces.md:8]: talked a bit with @andreibancioiu and will probably be segregated in a follow up PR.

[the remark this answers is not in the document]
Comment (andreibancioiu) [reply sdk-wallet/interfaces.md:8]: Indeed, we'll have: `ISigner`, `IVerifier`, `IKeysGenerator`, and, as now, `IMnemonicComputer`.

**Pull request merged**